stub/src/main/java/io/grpc/kotlin/CoroutineContextServerInterceptor.kt

```diff
@@ -49,8 +54,15 @@ abstract class CoroutineContextServerInterceptor : ServerInterceptor {
     call: ServerCall<ReqT, RespT>,
     headers: Metadata,
     next: ServerCallHandler<ReqT, RespT>
-  ): ServerCall.Listener<ReqT> =
-    withGrpcContext(GrpcContext.current().extendCoroutineContext(coroutineContext(call, headers))) {
+  ): ServerCall.Listener<ReqT> {
+    val coroutineContext = try {
+      coroutineContext(call, headers)
+    } catch (e: StatusException) {
+      call.close(e.status, Metadata())
```

Comment:
Fixed to use the trailers from the exception and added an appropriate test.

The problem which I encountered with the existing logic was that an exception thrown in the interceptor would bubble out and end up in the logs, where it could trigger an alert. In this case, the interceptor is detecting an error in client input and preventing further processing of the request. In such cases, we don't want the exception to show up in the logs but rather that an appropriate error be returned to the client.

Should there be another way to solve this without making any changes in the library, I'd be happy to use that -- I just couldn't find one.